cdap-common/src/main/java/io/cdap/cdap/common/internal/remote/TaskWorkerHttpHandlerInternal.java

```diff
@@ -169,31 +166,49 @@ private void enablePeriodicRestart(CConfiguration cConf,
         Constants.TaskWorker.CONTAINER_KILL_AFTER_DURATION_SECOND, 0);
     int lowerBound = (int) (duration - duration * DURATION_FRACTION);
     int upperBound = (int) (duration + duration * DURATION_FRACTION);
-    if (lowerBound > 0) {
-      int waitTime =
-          (new Random()).nextInt(upperBound - lowerBound) + lowerBound;
-      Executors.newSingleThreadScheduledExecutor(
-              Threads.createDaemonThreadFactory("task-worker-restart"))
-          .scheduleWithFixedDelay(
-              () -> {
-                if (mustRestart.get()) {
-                  // We force pod restart as the ongoing request has not finished since last
-                  // periodic restart check.
-                  stopper.accept("");
-                  return;
-                }
-                // we restart once ongoing request (which has set runningRequestCount to 1)
-                // finishes.
-                mustRestart.set(true);
-                if (runningRequestCount.compareAndSet(0, 1)) {
-                  // there is no ongoing request. pod gets restarted.
-                  stopper.accept("");
-                }
-              },
-              waitTime,
-              waitTime,
-              TimeUnit.SECONDS);
+
+    if (lowerBound <= 0) {
```

Comment:
Why? did you mean to check the duration?

Reply:
I think this was done for early exit in terms of code readability ? 

Previously it was. 
```
if (lowerBound > 0) {
  //logic 
 }
//Function ends return
} 
```

Now it's 
```
if (lowerBound <= 0) {
  return
 }
 
//logic
//Function ends return
} 
```

Reply:
Can you change it to duration? I mean, it looks equivalent (lowerBound = duration*0.9, so lowerBound <= 0 same as duration <= 0), but it's harder to understand the logic of why are we checking the calculated value.

Reply:
done